sqlglot/lineage.py

```diff
@@ -204,13 +207,15 @@ def to_node(
             source = scope.sources.get(table)
 
             if isinstance(source, Scope):
+                reference = dict(scope.references).get(table)
```

Comment:
i assume references is a list of tuples because there could be duplicates, this would essentially clobber the duplicates, but it's just for metadata i guess, maybe just do a for loop here instead of creating the dict?

reference if next(... , None)

Reply:
I found selected_sources and it seemed like a better option than references.